keps/sig-network/3070-reserved-service-ip-range/README.md

```diff
@@ -0,0 +1,456 @@
+guaranteeing full compatibility and no risk rolling back the feature.
+
+### Version Skew Strategy
+
+Same as with upgrade, there is no change in the bitmaps, keeping the feature 100% compatible.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ServiceIPStaticSubrange
+  - Components depending on the feature gate: kube-apiserver
+- [x] Other
+  - Describe the mechanism:
+  - The feature require to restart the apiserver to use the new allocation strategy for Services, however,
+  it will not require downtime in multi-control-plane environments.
+    
+###### Does enabling the feature change any default behavior?
+
+Dynamic allocated IPs will not be chosen from the first X values of the Service IP range, where X is obtained from the formula
+described in [Proposed Services ClusterIPs allocation model](#proposed-services-clusterips-allocation-model)
+This change is transparent to the user, since the IP is and was already random, it doesn't matter from which range the IP is obtained.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing, this only affects new requests and how these new request obtain the dynamically assigned IPs.
+
+###### Are there any tests for feature enablement/disablement?
+
+No need to add new test, current tests should keep working as today.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Current Services will not be affected, it doesn't matter in which range their IPs are allocated.
+This only affects the creation of new Services without an IP specified, there is no risk of impact on running workloads.
+
+###### What specific metrics should inform a rollback?
+
+The allocation logic already has the following metrics, that will be expanded with a new label containing information about the
+type of allocation requested (dynamic or static):
+
+  - allocated_ips
+  - available_ips
+  - allocation_total
+  - allocation_errors_total
+
+The increase of the errors metrics or the trend of the allocated_ips and available_ips doesn't change and new services are created or deleted.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+### Monitoring Requirements
+
+Current allocator metrics will be expanded with a new label to each metric containing information about the
+type of allocation requested (dynamic or static):
+
+  - allocated_ips
+  - available_ips
```

Comment:
@wojtek-t these 2 metrics can not track the information of the type of allocation, because when an ip is released we don't have that information.
however, the counters can have it, because are only incremented
```
  - allocation_total
  - allocation_errors_total
```

Reply:
Fair - that sounds good enough to me.

Can you clarify that here?